rx-java/rxgrpc-test/src/test/java/com/salesforce/rxgrpc/ShareIntegrationTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ *  Copyright (c) 2017, salesforce.com, inc.
+ *  All rights reserved.
+ *  Licensed under the BSD 3-Clause license.
+ *  For full license text, see LICENSE.txt file in the repo root  or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+package com.salesforce.rxgrpc;
+
+import com.google.common.collect.Lists;
+import com.salesforce.grpc.testing.contrib.NettyGrpcServerRule;
+import io.reactivex.BackpressureStrategy;
+import io.reactivex.Flowable;
+import io.reactivex.Observable;
+import io.reactivex.Single;
+import io.reactivex.flowables.ConnectableFlowable;
+import io.reactivex.observers.TestObserver;
+import org.junit.Rule;
+import org.junit.Test;
+
+import java.util.ArrayList;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Test to demonstrate splitting the output of an RxGrpc stream in RxJava.
+ * See: https://github.com/salesforce/reactive-grpc/issues/131
+ */
+public class ShareIntegrationTest {
+    @Rule
+    public NettyGrpcServerRule serverRule = new NettyGrpcServerRule();
+
+    @Test
+    public void serverShareShouldWork() {
+        RxGreeterGrpc.GreeterImplBase svc = new RxGreeterGrpc.GreeterImplBase() {
+            @Override
+            public Single<HelloResponse> sayHelloReqStream(Flowable<HelloRequest> rxRequest) {
+//                ConnectableFlowable<HelloRequest> shared = rxRequest
+//                        .doOnSubscribe(x -> System.out.println("SUBSCRIBE"))
+//                        .publish();
+
+                Flowable<HelloRequest> shared = rxRequest
+                        .doOnSubscribe(x -> System.out.println("SUBSCRIBE"))
+                        .share();
```

Comment:
@OlegDokuka This seems strange. If I use the `publish()/connect()` calls manually, the tests pass. If I use `share()` instead, `subscribe()` is called twice and an error occurs because we can't subscribe to an active gRPC call multiple times.

`share()` should work, since that's its purpose. Any idea why it doesn't?